fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/patient/ChunkReader.java

```diff
@@ -108,73 +126,88 @@ public ChunkReader() {
-                            continue;
-                        }
-                        try {
-                            FHIRGenerator.generator(Format.JSON).generate(res2, chunkData.getBufferStream());
-                            chunkData.getBufferStream().write(Constants.NDJSON_LINESEPERATOR);
-                            resSubTotal++;
-                        } catch (FHIRGeneratorException e) {
-                            if (res.getId() != null) {
-                                logger.log(Level.WARNING, "fillChunkDataBuffer: Error while writing resources with id '"
-                                        + res.getId() + "'", e);
-                            } else {
-                                logger.log(Level.WARNING,
-                                        "fillChunkDataBuffer: Error while writing resources with unknown id", e);
+
+                for (String patientId : patientIds) {
+
+                    searchContext = SearchUtil.parseQueryParameters("Patient", patientId,
+                            ModelSupport.getResourceType(resourceTypes.get(indexOfCurrentResourceType)), queryParameters, true);
+                    do {
+                        searchContext.setPageSize(pageSize);
+                        searchContext.setPageNumber(compartmentPageNum);
+                        FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
+                        txn.enroll();;
+                        FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
+                        List<Resource> resources = fhirPersistence.search(persistenceContext, resourceType).getResource();
+                        txn.unenroll();;
+                        compartmentPageNum++;
+
+                        for (Resource res : resources) {
+                            if (res == null || (isDoDuplicationCheck && loadedResourceIds.contains(res.getId()))) {
+                                continue;
+                            }
+                            try {
+                                FHIRGenerator.generator(Format.JSON).generate(res, chunkData.getBufferStream());
+                                chunkData.getBufferStream().write(Constants.NDJSON_LINESEPERATOR);
+                                resSubTotal++;
+                                if (isDoDuplicationCheck) {
+                                    loadedResourceIds.add(res.getId());
+                                }
+                            } catch (FHIRGeneratorException e) {
+                                if (res.getId() != null) {
+                                    logger.log(Level.WARNING, "fillChunkDataBuffer: Error while writing resources with id '"
+                                            + res.getId() + "'", e);
+                                } else {
+                                    logger.log(Level.WARNING,
```

Comment:
same comment as prior one - operationaloutcome / warning

Reply:
export doesn't require operation outcome in fhir spec, and this is why I didn't add like what was did in import, so just logging ...